### PR TITLE
Accumulo use a thread pool when doing batch scans

### PR DESCRIPTION
**Pull request by joeferner**
Benchmark https://github.com/visallo/vertexium/compare/fast-batch-reader?expand=1#diff-7de8e1270d184f1d847439d3722b3a36R9015

Before change: 36.852s
After change: 24.665s

This change also alleviates the strict need to close the iterator.

### CODE REVIEW

File: accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class ConnectorUtils {
+    public static ClientContext getContext(Connector connector) {
+        if (!(connector instanceof ConnectorImpl)) {
+            throw new VertexiumException("Invalid connector type. Expected " + ConnectorImpl.class.getName() + " found " + connector.getClass().getName());
+        }
+        try {
+            Field f = ConnectorImpl.class.getDeclaredField("context");
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: This won't be necessary when we upgrade to Accumulo 2.0. They seem to have added a get method: https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/clientImpl/ConnectorImpl.java#L74

File: accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            throw new VertexiumException("Could not get context field", e);
+        }
+    }
+
+    public static String getTableId(Connector connector, String tableName) throws TableNotFoundException {
+        if (!(connector instanceof ConnectorImpl)) {
+            throw new VertexiumException("Invalid connector type. Expected " + ConnectorImpl.class.getName() + " found " + connector.getClass().getName());
+        }
+        try {
+            Method m = ConnectorImpl.class.getDeclaredMethod("getTableId", String.class);
```
Comment (mwizeman) [inline accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: Looks like you might be able to use this helper instead: https://github.com/apache/accumulo/blob/rel/1.9.3/core/src/main/java/org/apache/accumulo/core/client/impl/Tables.java#L68

Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: I saw that, it's takes an `instance` which we don't keep around. Looks like in Accumulo 2.0 we can get that too... https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/clientImpl/ConnectorImpl.java#L96 (context comes from `getAccumuloClient()`)

File: accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java (hunk 1 above)
Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: Excellent

File: accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java (hunk 2 above)
Comment (mwizeman) [reply accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: I think you should be able to ask the `Connector` for its `Instance`: https://github.com/apache/accumulo/blob/rel/1.9.3/core/src/main/java/org/apache/accumulo/core/client/Connector.java#L216

Comment (joeferner) [reply accumulo/src/main/java/org/vertexium/accumulo/util/ConnectorUtils.java]: done

**Pull request merged**